File: algorithm/백준/5612.py
T = int(input())
N = int(input())
r = 0
for t in range(T):
    a, b = map(int, input().split())
    N += a - b
    if N < 0:
        r = 0
        break
    r = max(N, r)
print(r)

File: algorithm/백준/5612.cpp
#include <stdio.h>
int n, m;
int a, b;
int max;
int main() {
	scanf("%d", &n);
	scanf("%d", &m);
	max = m;

	while (n--) {
		scanf("%d %d", &a, &b);
		m += a - b;
		if (m < 0) {
			break;
		}
		if (max < m)
			max = m;
	}
	if (m < 0) {
		printf("0");
	}
	else {
		printf("%d", max);
	}
	return 0;
}